vivisect/analysis/generic/symswitchcase.py

```diff
@@ -0,0 +1,1307 @@
+
+    def iterCases(self):
+        '''
+        Generator which yields the case index and the target address
+        '''
+        vw = self.vw
+
+        (csemu, cseffs), aspath, fullpath = self.getSymbolikParts()
+
+        idxtype, symidx = self.getSymIdx()
+        logger.info('getSymIdx: %r/%r (0x%x)', idxtype, symidx, self.jmpva)
+
+        jmptgt = self.getSymTarget()
+        lower, upper, offset = self.getBounds()
+        logger.debug( " itercases: %r %r %r %r (0x%x)", symidx, lower, upper, offset, self.jmpva)
+
+        # use a TrackingSymbolikEmulator so we get the "readSymMemory()" function actually hitting 
+        # the r/o data from our VivWorkspace
+        symemu = TrackingSymbolikEmulator(vw)
+        symemu.setSymSnapshot(csemu.getSymSnapshot())
+
+        logger.debug("jmpva: 0x%x\t\tsymidx: %r", self.jmpva, symidx)   # TODO: we return either a Var name string or a SymObj (for Mem types)... should we just stick with the symbolik object?  since we're removing the ties to getSymVariable() and moving to replaceObj() and our nifty 'jmpidx' variable.
+        if idxtype == SYMT_VAR:
+            symidx = Var(symidx, self.vw.psize)
+        replaceObj(jmptgt, symidx, Var('jmpidx', symidx.getWidth()))
+        #workJmpTgt = jmptgt.update(emu=symemu)
+
+
+        for idx in range(lower-offset, upper-offset+1):
+            symemu.setSymVariable('jmpidx', Const(idx, 8))
+            workJmpTgt = jmptgt.update(emu=symemu)  # would "jmptgt.solve(vals={'jmpidx': idx})" work?
+            logger.info(" itercases: workJmpTgt: %r (0x%x)", workJmpTgt, self.jmpva)
+            coderef = workJmpTgt.solve(emu=symemu, vals={symidx:idx})
+            #coderef = jmptgt.update(emu=symemu)
+            logger.debug(" itercases: %r, %r", idx, hex(coderef))
+            yield idx + offset, coderef
+
+def makeNames(vw, jmpva, cases, baseoff=0):
+    '''
+    Create names and xrefs for each identified case.
+    '''
+    for addr, indexes in cases.items():
+        outstrings = []
+        combined = False
+        start = last = UNINIT_CASE_INDEX
+        for case in indexes:
+            if case == last+1:
+                if not combined:
+                    combined = True
+                    start = last
+            else:
+                if combined:
+                    combined = False
+                    outstrings.append("%Xto%X" % (start+baseoff, last+baseoff))
+                elif last != UNINIT_CASE_INDEX:
+                    outstrings.append("%X" % (last+baseoff))
+            last = case
+
+        # catch the last one if highest cases are combined
+        if combined:
+            combined = False
+            outstrings.append("%Xto%X" % (start+baseoff, last+baseoff))
+        else:
+            outstrings.append("%X" % (case+baseoff))
+
+        logger.info("OUTSTRINGS: %s (0x%x)", repr(outstrings), jmpva)
+
+        idxs = '_'.join(outstrings)
+        casename = "case_%s_%x" % (idxs, addr)
```

Comment:
We could end up with massive names here couldn't we?

Reply:
yes.  they're not as bad as they could be, but they can be kinda big.  but that's the name of the game.  i'm open to suggestions.
i'm already collapsing sequential options into ranges.